connect/runtime/src/test/java/org/apache/kafka/connect/util/TopicAdminTest.java

```diff
@@ -97,12 +103,65 @@ public void shouldNotCreateTopicWhenItAlreadyExists() {
     }
 
     @Test
-    public void shouldCreateTopicWhenItDoesNotExist() {
-        NewTopic newTopic = TopicAdmin.defineTopic("myTopic").partitions(1).compacted().build();
-        Cluster cluster = createCluster(1);
-        try (MockAdminClient mockAdminClient = new MockAdminClient(cluster.nodes(), cluster.nodeById(0))) {
-            TopicAdmin admin = new TopicAdmin(null, mockAdminClient);
-            assertTrue(admin.createTopic(newTopic));
+    public void shouldCreateTopicWithPartitionsWhenItDoesNotExist() {
+        for (int numBrokers = 1; numBrokers < 10; ++numBrokers) {
+            int expectedReplicas = Math.min(3, numBrokers);
+            int maxDefaultRf = Math.min(numBrokers, 5);
+            for (int numPartitions = 1; numPartitions != 30; ++numPartitions) {
+                NewTopic newTopic = TopicAdmin.defineTopic("myTopic").partitions(numPartitions).compacted().build();
+
+                // Try clusters with no default replication factor or default partitions
+                assertTopicCreation(numBrokers, newTopic, null, null, expectedReplicas, numPartitions);
+
+                // Try clusters with different default partitions
+                for (int defaultPartitions = 1; defaultPartitions != 20; ++defaultPartitions) {
+                    assertTopicCreation(numBrokers, newTopic, defaultPartitions, null, expectedReplicas, numPartitions);
+                }
+
+                // Try clusters with different default replication factors
+                for (int defaultRF = 1; defaultRF != maxDefaultRf; ++defaultRF) {
```

Comment:
I didn't comment elsewhere, but that's where `!=` has the potential of stackoverflow vs `<=`. I understand we trust ourselves to set `maxDefaultRf` to be `>0` but that's a couple indirections away.